Source/Core/DolphinQt/Settings/USBDeviceAddToWhitelistDialog.cpp

```diff
@@ -173,4 +179,43 @@ void USBDeviceAddToWhitelistDialog::OnDeviceSelection()
       device.split(QString::fromStdString(":"), QString::SplitBehavior::KeepEmptyParts)[1]);
   device_vid_textbox->setText(*vid);
   device_pid_textbox->setText(*pid);
+  // add_button->setEnabled(true); // the way selection works currently makes this unnecessary
+}
+
+void USBDeviceAddToWhitelistDialog::OnVIDTextboxTextChanged()
+{
+  QString* vid_text = new QString(device_vid_textbox->text());
```

Comment:
Why dynamic allocation? Seems like you're leaking memory too.

Could just be `const QString&` (preferably, to avoid a copy) or `QString`. Same in the other new method.

Reply:
Yes oops, it's a leftover when I was trying to get this line to build, turned out it badly wanted `text()` instead of `text` (I'm learning C++ along the way, you know ;))